Sources/URLNavigator.swift

```diff
@@ -117,11 +117,12 @@ open class URLNavigator {
   /// Returns a matched view controller from a specified URL.
   ///
   /// - parameter url: The URL to find view controllers.
+  /// - parameter userInfo: The user extra parameters you want add.
   /// - returns: A match view controller or `nil` if not matched.
-  open func viewController(for url: URLConvertible) -> UIViewController? {
+  open func viewController(for url: URLConvertible, with userInfo: [AnyHashable: Any]? = nil) -> UIViewController? {
```

Comment:
I think we don't need the parameter name `with`. It is OK to use just `userInfo` :smile: